src/routes/login.tsx

```diff
@@ -87,13 +87,17 @@ export function LoginForm() {
   }, [currentRole, currentType]);
 
   useLayoutEffect(() => {
-    if (isAuthenticated && search?.redirect) {
-      router.history.push(search.redirect);
-    } else if (isAdmin) {
-      router.history.push('/admin/xpub');
-    } else if (isUser) {
-      router.history.push('/user/access-keys');
+    if (!isAuthenticated) {
+      return;
     }
+
+    let redirect = search?.redirect?.includes('user') ? search?.redirect : '/user/access-keys';
+
+    if (isAdmin) {
+      redirect = search?.redirect?.includes('admin') ? search?.redirect : '/admin/xpub';
```

Comment:
What if `search?.redirect` is null/undefined?

Reply:
It will return the default path 
1. /users/access-keys
2. or if IsAdmin -> /admin/xpub

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining